src/vstt/stats.py

```diff
@@ -493,3 +524,39 @@ def preprocess_mouse_positions(mouse_positions: np.ndarray) -> np.ndarray:
         mouse_positions.reshape(0, 2) if mouse_positions.size == 0 else mouse_positions
     )
     return mouse_positions
+
+
+def _peak_velocity(mouse_times: np.ndarray, mouse_positions: np.ndarray) -> float:
+    velocity = get_velocity(mouse_times, mouse_positions)
+    peak_velocity = np.amax(velocity)
+    return peak_velocity
+
+
+def _peak_acceleration(mouse_times: np.ndarray, mouse_positions: np.ndarray) -> float:
+    acceleration = get_acceleration(mouse_times, mouse_positions)
+    peak_acceleration = np.amax(acceleration)
+    return peak_acceleration
+
+
+def get_derivative(y: np.ndarray, x: np.ndarray) -> np.ndarray:
+    dy_dx = np.diff(y) / np.diff(x)
+    return dy_dx
+
+
+def get_velocity(mouse_times: np.ndarray, mouse_positions: np.ndarray) -> np.ndarray:
+    if mouse_times.size == 0 or mouse_positions.size == 0:
+        return np.array([0])
+    first_order_derivative = get_derivative(mouse_positions.transpose(), mouse_times)
+    velocity = LA.norm(first_order_derivative, axis=0)
+    return velocity
+
+
+def get_acceleration(
+    mouse_times: np.ndarray, mouse_positions: np.ndarray
+) -> np.ndarray:
+    if mouse_times.size == 0 or mouse_positions.size == 0:
+        return np.array([0])
+    first_order_derivative = get_derivative(mouse_positions.transpose(), mouse_times)
+    second_order_derivative = get_derivative(first_order_derivative, mouse_times[:-1])
```

Comment:
You checked earlier that `mouse_times` is not empty, but if it has exactly 1 element then I think this expression will give an error - could you add a test with a single mouse_time? (also maybe it would be more robust to move the checks for empty arrays to the `get_derivative` function?)